src/extensions/manager/ExtensionCard.tsx

```diff
@@ -22,35 +23,47 @@ export class ExtensionCard<S extends ConfigurationSubject, C extends Settings> e
 > {
     public render(): JSX.Element | null {
         const { node, ...props } = this.props
+        const manifest: ExtensionManifest | undefined =
+            node.manifest && !isErrorLike(node.manifest) ? node.manifest : undefined
+
         return (
             <div className="d-flex col-sm-6 col-md-6 col-lg-4 pb-4">
                 <div className="extension-card card">
                     <LinkOrSpan
                         to={node.registryExtension && node.registryExtension.url}
                         className="card-body extension-card-body d-flex flex-column"
                     >
-                        <h4 className="card-title extension-card-body-title mb-0">
-                            {node.manifest && !isErrorLike(node.manifest) && node.manifest.title
-                                ? node.manifest.title
-                                : node.id}
-                        </h4>
-                        <div className="extension-card-body-text d-inline-block mt-1">
-                            {node.manifest ? (
-                                isErrorLike(node.manifest) ? (
-                                    <span className="text-danger small" title={node.manifest.message}>
-                                        <props.extensions.context.icons.Warning className="icon-inline" /> Invalid
-                                        manifest
-                                    </span>
-                                ) : (
-                                    node.manifest.description && (
-                                        <span className="text-muted">{node.manifest.description}</span>
-                                    )
-                                )
-                            ) : (
-                                <span className="text-warning small">
-                                    <props.extensions.context.icons.Warning className="icon-inline" /> No manifest
-                                </span>
-                            )}
+                        <div className="d-flex">
+                            {manifest &&
+                                manifest.icon &&
+                                new URL(manifest.icon).protocol === 'data:' &&
+                                /^data:image\/png/.test(manifest.icon) && (
```

Comment:
I'm blocking everything but `data:image/png` URIs to sidestep the complexity of SVG sanitization, avoid leaking referrer, and reduce the jankiness of images loading at random times.

It's awkward, but you can generate a data URI from a PNG like this:

```
$ printf "data:image/png;base64," > uri.txt ; cat icon.png | base64 >> uri.txt
```

Then copy the content of `uri.txt` into the `"icon": "..."` field in `package.json`.